build-analyze.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# Configuration for SonarCloud
+SONAR_HOST_URL=https://sonarcloud.io
+#SONAR_TOKEN= # access token from SonarCloud projet creation page -Dsonar.login=XXXX: define in the environement through the CI
+export SONAR_SCANNER_VERSION="4.6.1.2450" # Find the latest version in the "Windows" link on this page:
+                                          # https://sonarcloud.io/documentation/analysis/scan/sonarscanner/
+export BUILD_WRAPPER_OUT_DIR="build_wrapper_output_directory" # Directory where build-wrapper output will be placed
+
+mkdir $HOME/.sonar
+export SONAR_SCANNER_HOME=$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux
```

Comment:
IMO this extra variable definition does not help readability as it is far from its use and it is used only once. I would inline it in the point of use (on l. 21):
```
export PATH=$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux/bin:$PATH
```